### DISCUSSION TITLE
My expectation as desktop designer for Akira

### DISCUSSION THREAD

**Discussion by albfan** [akiraux/Akira#514] [Ideas]
This screencast represent what looking for as an Akira user:

![pencil screencast](https://user-images.githubusercontent.com/220968/51612235-501d7380-1f21-11e9-8c83-124ddb4f5c1d.png)

which result in:

![gitg ini editor](https://user-images.githubusercontent.com/220968/51612236-50b60a00-1f21-11e9-9876-b5385b6c39cf.png)

It should take like 5 min to design this, but there lot of tweaks, copy/paste from printScreen, shrink s involved which make it take lot more time. I can add a screencast with problems we can solve on Akira about design this kind of UI interfaces

I will love to make it interactive (i.e. "Color..." button launch the color chooser dialog)

This is related with https://gitlab.gnome.org/GNOME/gitg/issues/180

**Comment by Alecaddd** [comment]
This is an example for reusable assets and templates library, right?

If that's the case, I'd suggest not to focus right now on this as I don't think we'll be able to ship it with version 1.
Reusable assets and a built-in library is definitely something we want to have.
I' not sure about the interactivity of those templates.

I'd suggest to leave this issue open for future reference and use it as a discussion room to define how we want to implement this once v1 has been shipped.

**Comment by albfan** [comment]
Yes, totally. That's the intention. Just to be clear that Akira may do something like that.

I guess [milestone v1](https://github.com/Alecaddd/Akira/milestone/2) will be related to object creation, transformation and alignment, right?

**Comment by Alecaddd** [comment]
Yes, that's what I originally had in mind.
Milestone 1 will be mostly focused on nailing down the canvas and offering all the common features of a standard design tool, like you said, creation, transformation, alignment, etc.

Great stuff nonetheless, this is an awesome objective to keep in mind for future releases, and immensely useful for designers.

**Comment by julientaq** [comment]
When we'll talk about the component library, there is something pretty important: compenent have states, and u should be able to switch from one state to another while overriding some off what's inside.
